infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -97,25 +100,76 @@ public LogReplicationMetadataManager(CorfuRuntime runtime, LogReplicationContext
+     * @param metadataClass - Protobuf generated Metadata class (can be null)
+     * @return - a generic table type.
+     */
+    public static Table tryOpenTable(CorfuStore corfuStore, String namespace, String tableName,
+                                     Class keyClass, Class valueClass, Class metadataClass) {
+        Table table = null;
         try {
-            this.metadataTable = this.corfuStore.openTable(NAMESPACE, METADATA_TABLE_NAME,
-                    LogReplicationSession.class, ReplicationMetadata.class, null,
-                    TableOptions.fromProtoSchema(ReplicationMetadata.class));
-
-            this.statusTable = this.corfuStore.openTable(NAMESPACE, LogReplicationUtils.REPLICATION_STATUS_TABLE_NAME,
-                    LogReplicationSession.class, ReplicationStatus.class, null,
-                    TableOptions.fromProtoSchema(ReplicationStatus.class));
-
-            this.replicationEventTable = this.corfuStore.openTable(NAMESPACE, REPLICATION_EVENT_TABLE_NAME,
-                    ReplicationEventInfoKey.class,
-                    ReplicationEvent.class,
-                    null,
-                    TableOptions.fromProtoSchema(ReplicationEvent.class));
-        } catch (Exception e) {
-            log.error("Caught an exception while opening metadata tables", e);
-            throw new ReplicationWriterException(e);
+            table = corfuStore.openTable(namespace, tableName,
+                    keyClass, valueClass, metadataClass,
+                    TableOptions.fromProtoSchema(valueClass));
+        } catch (Exception ee) {
+            log.error("Caught an exception while opening "+namespace+"$"+tableName, ee);
+            throw new ReplicationWriterException(ee);
         }
+        return table;
+    }
+
+    /**
+     * The following tables need to move from old type to new type
+     * 1. CORFU-REPLICATION-WRITER-UUID-OF-CLUSTER -> LogReplicationMetadataTable
+     * Type: Schema change AND name change
+     *    (LogReplicationMetadataKey, LogReplicationMetadataVal) ->
+     *     (LogReplicationSession, ReplicationMetadata)
+     * Action: Open the old table with old types and Clear the old table
+     *         Not used in new code. Optional as failure to clear won't cause issues.
+     *         TODO: consider scanning registry for signature and wipe out contents.
+     *
+     * 2. LogReplicationStatus
+     * Type: Schema change for key AND value types with NO table name change
+     * (ReplicationStatusKey, ReplicationStatusVal) ->
+     * (LogReplicationSession.class, ReplicationMetadata.class)
+     * Action: Make serializer aware of old types, open table with new type, clear it
+     *
+     * 3. LogReplicationEventTable
+     * Type: Schema change for Key type only
+     * (ReplicationEventKey, ReplicationEvent) ->
+     * (ReplicationEventInfoKey, ReplicationEvent)
+     * Action: Make serializer aware of old key type, open table with new type, clear it
```

Comment:
Thanks for the detailed comment. It looks like addLegacyTypesToSerializer is adding the legacy types into serializer.

But, opening table with new type and clearing part is outside of addLegacyTypesToSerializer ? Little confused with comment and method definition

Reply:
once the `ProtobufSerializer` is made aware of the type using the new `addTypeToClassMap` opening that table later and doing operations won't hit SerializerExceptions because the serializer will know how to interpret the data it is seeing.